tests/test_views.py

```diff
@@ -0,0 +1,801 @@
+
+        request = rf.put('/', xml, content_type='application/xml', HTTP_HOST='example.com')
+        response = views.app_event(request, identifier)
+
+        assert response.get('Content-Type') == self.CONTENT_TYPE
+
+    def test_list_returns_ok(self, rf):
+        factories.EventFactory.create_batch(30)
+        request = rf.get('/')
+        response = views.app_event(request)
+        assert response.status_code == 200
+
+    def test_list_number_of_events(self, rf):
+        factories.EventFactory.create_batch(self.RESULTS_PER_PAGE * 2)
+        request = rf.get('/')
+        response = views.app_event(request)
+        xml = objectify.fromstring(response.content)
+
+        assert len(xml.entry) == self.RESULTS_PER_PAGE
+
+    @pytest.mark.xfail(reason='Global name DATE_FORMAT is not defined.')
+    def test_list_filtering_by_start_date(self, rf):
+        datetime_obj = timezone.now().replace(2015, 1, 1)
+        factories.EventFactory.create_batch(30, event_date_time=datetime_obj)
+        event = factories.EventFactory.create(event_date_time=timezone.now())
+
+        request = rf.get('/?start_date=01/31/2015')
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
+    @pytest.mark.xfail(reason='Global name DATE_FORMAT is not defined.')
+    def test_list_filtering_by_end_date(self, rf):
+        datetime_obj = timezone.now().replace(2015, 1, 1)
+        factories.EventFactory.create_batch(30, event_date_time=timezone.now())
+        event = factories.EventFactory.create(event_date_time=datetime_obj)
+
+        request = rf.get('/?end_date=01/31/2015')
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
+    def test_list_filtering_by_linking_object_id(self, rf):
+        factories.EventFactory.create_batch(30)
+        event = factories.EventFactory.create(linking_objects=True)
+        linking_object = event.linking_objects.first()
+
+        url = '/?link_object_id={0}'.format(linking_object.object_identifier)
+        request = rf.get(url)
+        response = views.app_event(request)
+
+        assert self.event_in_filtered_results(response, event)
+
+    def test_list_filtering_by_event_outcome(self, rf):
+        factories.EventFactory.create_batch(30)
+
+        event_outcome = 'success'
+        event = factories.EventFactory.create(event_outcome=event_outcome)
+
+        request = rf.get('?outcome={0}'.format(event_outcome))
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
+    def test_list_filtering_by_event_type(self, rf):
+        factories.EventFactory.create_batch(30)
+
+        event_type = 'Test Type'
+        event = factories.EventFactory.create(event_type=event_type)
+
+        request = rf.get('?type={0}'.format(event_type))
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
```

Comment:
There is not a test here for a GET request with no identifier related to using the `orderby` parameter. I opened up an issue related to a case where it will throw a 500 error. https://github.com/unt-libraries/django-premis-event-service/issues/24.